### PR TITLE
Updates Poncho and doc to reflect Conda's specification format

### PR DESCRIPTION
**Pull request by BarrySlyDelgado**
This addresses #2949. the key "packages" is changed to "dependencies". pip packages are under "dependencies" as well within their own dict. The doc manual has been updated to reflect this change.

### CODE REVIEW

**Comment by dthain** [comment]
Will this break existing tools (Parsl, Coffea, TopEFT) that are already using Poncho?

**Comment by BarrySlyDelgado** [comment]
@dthain This would affect tools that use  `poncho_package_create`  Already created specifications would have to be updated to the new format to use poncho_package_create. So, I believe that topEFT would have to be modified.

**Comment by dthain** [comment]
So we can't just break our most important customers.
And we must consider the case that multiple versions will exist simultaneously.
The way to proceed here is to permit the new keywords while retaining backwards compatibility with the old ones.
Then, update Parsl/Coffea to use the new keywords.
Finally, remove support for the old keywords once everyone has transitioned.

**Comment by BarrySlyDelgado** [comment]
I will change it to allow both formats for the specifications.

**Comment by BarrySlyDelgado** [comment]
Poncho_package_create now accepts both specification formats. All existing tools that use Poncho should still operate correctly.

File: doc/manuals/poncho/index.md [hunk 1]
```diff
@@ -57,14 +63,11 @@ Then to create a complete package from the specification:
...
 
-The outputed tarball will be put into a directory named `envs` if a specific 
-cache directory is not specified.
-
 Once created, this package can be moved to another machine for execution.
 Then, to run a program in the environment:
 
 ```
-poncho_package_run -e package.tar.gz -- example.py
+poncho_package_run -e package.tar.gz python example.py
```
Comment (btovar) [inline doc/manuals/poncho/index.md:70]: Let's keep the -- here, e.g.: `poncho_package_run -e package.tar.gz -- python example.py`

**Review by btovar — CHANGES_REQUESTED** [review]

**Review by btovar — APPROVED** [review]

**Pull request merged**